packages/js/settings-editor/typings/global.d.ts

```diff
@@ -0,0 +1,53 @@
+type SettingField = {
+    title?: string;
+    type: string;
+    id: string;
+    desc?: string;
+    desc_tip?: boolean | string;
+    default?: string | number | boolean | object;
+    value: string | number | boolean | object;
+    placeholder?: string;
+    custom_attributes?: {
+        [key: string]: string | number;
+    };
+    options?: {
+		[key: string]: string;
+	};
+    css?: string;
+    class?: string;
+    checkboxgroup?: 'start' | 'end' | '';
+    autoload?: boolean;
+    show_if_checked?: string;
+    content?: string;
+    [key: string]: any;
+};
+
+type SettingsSection = {
+    label: string;
+    settings: SettingField[];
+};
+
+type SettingsPageData = {
+    label: string;
+    sections: {
+        [key: string]: SettingsSection;
```

Comment:
this will be an array

Reply:
Fixed. https://github.com/woocommerce/woocommerce/pull/52692/files#diff-3ac9817d2ebb47b820627e2ad030a3579e24a74925d7151e90a63e7ac5135629R27